File: bank-service/src/main/java/rs/edu/raf/service/impl/CurrencyServiceImpl.java
package rs.edu.raf.service.impl;

import lombok.AllArgsConstructor;
import org.springframework.stereotype.Service;
import rs.edu.raf.dto.CurrencyCreateDTO;
import rs.edu.raf.dto.CurrencyDTO;
import rs.edu.raf.exceptions.CurrencyNotFoundException;
import rs.edu.raf.mapper.CurrencyMapper;
import rs.edu.raf.model.Currency;
import rs.edu.raf.repository.CurrencyRepository;
import rs.edu.raf.service.CurrencyService;

import java.util.List;

@Service
@AllArgsConstructor
public class CurrencyServiceImpl implements CurrencyService {
    private CurrencyRepository currencyRepository;
    private CurrencyMapper currencyMapper;

    @Override
    public CurrencyDTO createCurrency(CurrencyCreateDTO currencyCreateDTO) {
        return currencyMapper.currencyToCurrencyDTO(currencyRepository.save(currencyMapper.currencyCreateDTOtoCurrency(currencyCreateDTO)));
    }

    @Override
    public List<CurrencyDTO> getAllCurrencies() {
        return currencyRepository.findAll().stream().map(currencyMapper::currencyToCurrencyDTO).toList();
    }

    @Override
    public void deleteCurrency(Long id) {
        Currency currency = currencyRepository.findById(id)
                .orElseThrow(()->new CurrencyNotFoundException(id));

        currencyRepository.delete(currency);
    }
}

File: bank-service-kotlin/src/main/kotlin/org/example/bankservicekotlin/service/impl/CurrencyServiceImpl.kt
package org.example.bankservicekotlin.service.impl

import org.example.bankservicekotlin.dto.CurrencyCreateDTO
import org.example.bankservicekotlin.dto.CurrencyDTO
import org.example.bankservicekotlin.exception.CurrencyNotFoundException
import org.example.bankservicekotlin.mapper.CurrencyMapper
import org.example.bankservicekotlin.model.Currency
import org.example.bankservicekotlin.repository.CurrencyRepository
import org.example.bankservicekotlin.service.CurrencyService
import org.springframework.stereotype.Service

@Service
class CurrencyServiceImpl(
    private val currencyRepository: CurrencyRepository,
    private val currencyMapper: CurrencyMapper
): CurrencyService {
    override fun createCurrency(currencyCreateDTO: CurrencyCreateDTO): CurrencyDTO {
        return currencyMapper.currencyToCurrencyDTO(
            currencyRepository.save(
                currencyMapper.currencyCreateDTOtoCurrency(
                    currencyCreateDTO
                )
            )
        )
    }

    override fun getAllCurrencies(): List<CurrencyDTO> {
        return currencyRepository.findAll().stream().map(currencyMapper::currencyToCurrencyDTO).toList()
    }

    override fun deleteCurrency(id: Long) {
        val currency: Currency = currencyRepository.findById(id)
            .orElseThrow { CurrencyNotFoundException(id) }

        currencyRepository.delete(currency)
    }
}